frontend/src/components/App/index.tsx

```diff
@@ -47,31 +49,51 @@ Font.register({
   ],
 });
 
-const Wrapper = () => {
+const Wrapper = memo(() => {
+  const userAuth = useSelector(userAuthSelector);
+
+  const isUserAuthenticated = useMemo(() => {
+    return userAuth !== undefined;
+  }, [userAuth]);
+
+  // renders the Auth modal only if a user isn't already authenticated
+  const renderedAuthModal = useMemo(() => {
+    if (isUserAuthenticated) {
+      return null;
+    }
+    return <AuthModal />;
+  }, [isUserAuthenticated]);
```

Comment:
Could this logic live in the `AuthModal` component instead?

Reply:
of course will change it.